internal/core/item_tree.rs

```diff
@@ -268,14 +268,16 @@ impl ItemRc {
         step_in: &dyn Fn(ItemRc) -> ItemRc,
         step_out: &dyn Fn(&crate::item_tree::ComponentItemTree, usize) -> Option<usize>,
     ) -> Self {
-        let comp_ref_pin = vtable::VRc::borrow_pin(&self.component);
-        let item_tree = crate::item_tree::ComponentItemTree::new(&comp_ref_pin);
+        let mut component = self.component().clone();
```

Comment:
there’s an extra clone when calling the function.

```suggestion
        let mut component = self.component.clone();
```